packages/graphback-runtime-knex/src/knexQueryMapper.ts

```diff
@@ -1,140 +1,168 @@
 import Knex from 'knex';
+import { QueryFilter, QueryFilterOperator } from '@graphback/core';
 
-interface OperatorMap {
-  ne: '<>',
-  eq: '=',
-  le: '<=',
-  lt: '<',
-  ge: '>=',
-  gt: '>',
-  contains: 'like',
-  startswith: 'like',
-  endswith: 'like',
-}
+const knexOperators = ['=', '<>', '<=', '<', '>', '>=', 'like', 'between', 'in'] as const;
+const knexMethods = ['where', 'orWhere', 'orWhereNot', 'whereNot'] as const;
 
-const methodMapping = {
-  in: 'whereIn',
-  between: 'whereBetween',
-  default: 'where'
-}
+type KnexQueryOperator = typeof knexOperators[number];
+type KnexMethod = typeof knexMethods[number];
+type KnexWhereConditionMap = [KnexQueryOperator, any];
+type RootQuerySelector = 'and' | 'or' | 'not';
 
-const notMethodMap = {
-  where: 'whereNot',
-  whereNull: 'whereNotNull',
-  between: 'whereNotBetween',
-  in: 'whereNotIn'
-}
+type KnexRootQuerySelectorBuilderFn = (builder: Knex.QueryBuilder, filter: QueryFilter | QueryFilter[], rootSelectorContext: RootQuerySelectorContext) => Knex.QueryBuilder;
+type KnexQueryBuilderMapFn = (builder: Knex.QueryBuilder, filter: QueryFilter | QueryFilter[]) => Knex.QueryBuilder;
 
-const AND_FIELD = 'and';
-const OR_FIELD = 'or';
-const NOT_FIELD = 'not';
-
-function mapOperator(operator: any) {
-  const operatorMap: OperatorMap = {
-    ne: '<>',
-    eq: '=',
-    le: '<=',
-    lt: '<',
-    ge: '>=',
-    gt: '>',
-    contains: 'like',
-    startswith: 'like',
-    endswith: 'like'
-  }
+interface RootQuerySelectorContext {
+  and?: boolean
+  or?: boolean
+  not?: boolean
+}
 
-  const oprLower = operator.toLowerCase()
+export interface CRUDKnexQueryMapper {
+  /**
+   * Maps a Graphback QueryFilter to a Knex query
+   * 
+   * @param {QueryFilter} [filter] - input filter
+   */
+  buildQuery(filter?: QueryFilter): Knex.QueryBuilder;
+}
 
-  if (!Object.keys(operatorMap).includes(oprLower)) {
-    throw Error(`Not supported operator: ${operator}`)
+const mapQueryFilterOperatorToKnexWhereCondition = (operator: QueryFilterOperator, value: any): KnexWhereConditionMap => {
+  const operatorToWhereConditionMap: { [key in QueryFilterOperator]: KnexWhereConditionMap } = {
+    eq: ['=', value],
+    ne: ['<>', value],
+    lt: ['<', value],
+    le: ['<=', value],
+    ge: ['>=', value],
+    gt: ['>', value],
+    contains: ['like', `%${value}%`],
+    startsWith: ['like', `${value}%`],
```

Comment:
Adding new operator ( e.g the one described in https://github.com/aerogear/graphback/issues/1967#issuecomment-685824275) would be as simple as adding the mapping here? 

Same question for Mongo query builder.

Reply:
Yes!

Reply:
It's much better already. Thank you.

When I was looking at the Mongo part it seemed doable also, not quite similar but it is simple enough.